pkg/cluster/kubernetes/upgrade_checks.go

```diff
@@ -0,0 +1,349 @@
+				case k8s.SchedulerID:
+					upgradeCheckMessage, checkErr := checkK8sComponentRemovedFlags(id, pod.Spec.Containers[0].Command, k8sComponentChecks.kubeSchedulerChecks.removedFlags)
+					if checkErr != nil {
+						return checkErr
+					}
+
+					upgradeCheckErrors = multierror.Append(upgradeCheckErrors, upgradeCheckMessage)
+				}
+
+				upgradeCheckMessage, err := checkK8sComponentRemovedFeatureGates(id, pod.Spec.Containers[0].Command, k8sComponentChecks.removedFeatureGates)
+				if err != nil {
+					return err
+				}
+
+				upgradeCheckErrors = multierror.Append(upgradeCheckErrors, upgradeCheckMessage)
+			}
+		}
+
+		if upgradeCheckErrors.ErrorOrNil() != nil {
+			upgradeCheckErrors.ErrorFormat = func(errs []error) string {
+				lines := []string{}
+
+				for _, err := range errs {
+					lines = append(lines, fmt.Sprintf(" %s", err.Error()))
+				}
+
+				count := color.RedString(pluralize.NewClient().Pluralize("error", len(lines), true))
+
+				return fmt.Sprintf("%s occurred:\n%s", count, strings.Join(lines, "\n"))
+			}
+
+			return upgradeCheckErrors
+		}
+	}
+
+	return nil
+}
+
+func checkK8sComponentRemovedFlags(component string, flags []string, removedFlags []string) (k8sUpgradeCheckErr, error) {
+	upgradeCheckMessages := bytes.NewBuffer([]byte{})
+
+	w := tabwriter.NewWriter(upgradeCheckMessages, 0, 0, 3, ' ', 0)
+
+	hasRemovedFlags := false
+
+	for _, removedFlag := range removedFlags {
+		if slices.Contains(flags, func(s string) bool {
+			return strings.HasPrefix(s, "--"+removedFlag)
+		}) {
+			if !hasRemovedFlags {
+				fmt.Fprintf(w, "\nCOMPONENT\tREMOVED FLAG\n")
+			}
+
+			hasRemovedFlags = true
+
+			fmt.Fprintf(w, "%s\t%s\n", component, removedFlag)
+		}
+	}
+
+	if err := w.Flush(); err != nil {
+		return nil, err
+	}
+
+	if hasRemovedFlags {
+		return errors.New(upgradeCheckMessages.String()), nil
+	}
+
+	return nil, nil
+}
+
+func checkK8sComponentRemovedFeatureGates(component string, flags []string, removedFeatureGates []string) (k8sUpgradeCheckErr, error) {
+	upgradeCheckMessages := bytes.NewBuffer([]byte{})
+
+	w := tabwriter.NewWriter(upgradeCheckMessages, 0, 0, 3, ' ', 0)
+
+	hasRemovedFeatureGates := false
+
+	featureGateFlags := slices.Filter(flags, func(s string) bool {
+		return strings.HasPrefix(s, "--feature-gates")
+	})
+
+	if len(featureGateFlags) > 0 {
+		featureGates := strings.Split(strings.TrimPrefix(featureGateFlags[0], "--feature-gates="), ",")
+
+		for _, removedFeatureGate := range removedFeatureGates {
+			slices.Contains(featureGates, func(s string) bool {
+				return removedFeatureGate == strings.Split(s, "=")[0]
+			})
+
+			if !hasRemovedFeatureGates {
+				fmt.Fprintf(w, "\nCOMPONENT\tREMOVED FEATURE GATE\n")
+			}
+
+			hasRemovedFeatureGates = true
+
+			fmt.Fprintf(w, "%s\t%s\n", component, removedFeatureGate)
+		}
+	}
+
+	if err := w.Flush(); err != nil {
+		return nil, err
+	}
+
+	if hasRemovedFeatureGates {
+		return errors.New(upgradeCheckMessages.String()), nil
```

Comment:
I think this could be done a bit cleaner way. 

We could declare a custom error type, like:

```
type RemoveFeatureGatesError struct {
   Items []struct{
      Component string
       FeatureGate string
   }
}

func (e *RemoveFeatureGatesError) Error() string{
   var buf bytes.Buffer

   w := tabwriter.NewWriter(buf, 0, 0, 3, ' ', 0)
   fmt.Fprintf(w, "\nCOMPONENT\tREMOVED FEATURE GATE\n")
   
   
    ....
    return buf.String()
}
```